# Praktikum Modul 2 AI
### Rehana Putri Saslabilla ( 5027221015 )

- Selamat datang di Praktikum Modul 2, bantu Rendy Atreides dalam menyelesaikan soal soal modul Statistika dan Probabilitas. <br>
- Jika ada pertanyaan silahkan tanyakan di **#use-ask** discord praktikum.

# Table of Contents
- [Case A](#A)
    - [ Soal 1](#1)
- [Case B](#B)
    - [ Soal 2](#2)
- [Case C](#C)
    - [ Soal 3](#3)
- [Case D](#D)
    - [ Soal 4](#4)
    - [ Soal 5](#5)
- [ Soal 6](#6)

In [ ]:
# Tambahkan sendiri seperlunya
import numpy as np
import pandas as pd

from scipy.stats import binom

# Diberikan data
n_flu_dan_vaksin = 200
n_flu_dan_tidak_vaksin = 900
n_tidak_flu_dan_vaksin = 4000
n_tidak_flu_dan_tidak_vaksin = 9000


<a name='A'></a>
## Case A

- Rendy Atreides mempunyai penglihatan terhadap musim hujan yang akan datang, karena hal tersebut ada kemungkinan/probabilitas teman dan kerabat nya untuk mendapatkan flu, $P(\text{flu})$
- Jika orang tersebut mendapatkan **vaksinasi**, maka peluang orang tersebut untuk terjangkit flu **seharusnya** berubah.

- Diberikan data berikut

| Kondisi   | Suhu Rata-rata (°C) | Vaksin | Tidak Vaksin | PDB Regional (dalam Miliar) | Jumlah Hari Libur Nasional | Rata-rata Jam Matahari per Hari |
|-----------|---------------------|--------|--------------|-----------------------------|----------------------------|---------------------------------|
| Flu       | 27                  | 200    | 900          | 650                         | 12                         | 6                               |
| Tidak Flu | 27                  | 4000   | 9000         | 650                         | 12                         | 6                               |




---
<a name='1'></a>
### **Soal 1**:
A. Dari data di atas, berapa peluang seseorang akan mengidap flu apabila telah vaksin? <br>

$$
\begin{align*}
P(\text{flu} \ | \ \text{vaksin}) &= ?
\end{align*}
$$

B. Dari data di atas, berapa peluang seseorang akan mengidap flu apabila **belum** divaksin?

In [ ]:
n_flu_dan_vaksin = 200
n_flu_dan_tidak_vaksin = 900

n_tidak_flu_dan_vaksin = 4000
n_tidak_flu_dan_tidak_vaksin = 9000

### A. Dari data di atas, berapa peluang seseorang akan mengidap flu apabila telah vaksin?

In [ ]:
n_vaksin = n_flu_dan_vaksin + n_tidak_flu_dan_vaksin

print(f"Jumlah orang yang divaksin : {n_vaksin}")

p_flu_jika_vaksin = n_flu_dan_vaksin / n_vaksin

print(f"A. Peluang orang mengalami flu jika sudah vaksin : {p_flu_jika_vaksin:.4f}")

Jumlah orang yang divaksin : 4200
A. Peluang orang mengalami flu jika sudah vaksin : 0.0476


Solver:

$$
\begin{align*}
P(\text{flu} \ | \ \text{vaksin}) &= \cfrac{n(\text{flu} \cap \text{vaksin})}{n(\text{vaksin})}
\end{align*}
$$

### B. Dari data di atas, berapa peluang seseorang akan mengidap flu apabila **belum** divaksin?

In [ ]:
n_tidak_vaksin = n_flu_dan_tidak_vaksin + n_tidak_flu_dan_tidak_vaksin

p_flu_jika_belum_vaksin = n_flu_dan_tidak_vaksin / n_tidak_vaksin
print(f"Peluang orang mengalami flu jika belum vaksin : {p_flu_jika_belum_vaksin:.4f}")

Peluang orang mengalami flu jika belum vaksin : 0.0909


## **Penjelasan Penyelesaian CASE A** <br>
Untuk menyelesaikan soal 1A dan 1B, menggunakan **Teorema Bayes**. Teorema Bayes digunakan untuk menghitung probabilitas posterior dari suatu kejadian berdasarkan informasi sebelumnya.

## ***Soal 1A : Peluang Mengidap Flu Setelah Divaksinasi***<br>
Untuk menghitung peluang seseorang mengidap flu setelah divaksinasi, langkah-langkahnya adalah sebagai berikut:

**Langkah-langkah:**
1. Menghitung jumlah orang yang telah divaksin ($n_{\text{vaksin}}$).
2. Menghitung jumlah orang yang mengalami flu dan telah divaksin ($n_{\text{flu dan vaksin}}$).
3. Menggunakan teorema Bayes untuk menghitung peluang seseorang akan mengidap flu apabila telah divaksin.

**Penyelesaian:**
- $n_{\text{vaksin}} = 200 + 4000 = 4200$
- $n_{\text{flu dan vaksin}} = 200$

Gunakan teorema Bayes:
$$
P(\text{flu} \ | \ \text{vaksin}) = \frac{n_{\text{flu dan vaksin}}}{n_{\text{vaksin}}}
$$

**Perhitungan:**
$$
\begin{align*}
P(\text{flu} \ | \ \text{vaksin}) &= \frac{n(\text{flu} \cap \text{vaksin})}{n(\text{vaksin})} \\
&= \frac{200}{4200} \\
&\approx 0.0476
\end{align*}
$$
<br>
## ***Soal 1B: Peluang Mengidap Flu Tanpa Vaksinasi***
Untuk menghitung peluang seseorang mengidap flu tanpa vaksinasi, langkah-langkahnya adalah sebagai berikut:

**Langkah-langkah:**
1. Menghitung jumlah orang yang belum divaksin ($n_{\text{tidak vaksin}}$).
2. Menghitung jumlah orang yang mengalami flu dan belum divaksin ($n_{\text{flu dan tidak vaksin}}$).
3. Menggunakan teorema Bayes untuk menghitung peluang seseorang akan mengidap flu apabila belum divaksin.

**Penyelesaian:**
- $n_{\text{tidak vaksin}} = 900 + 9000 = 9900$
- $n_{\text{flu dan tidak vaksin}} = 900$

Gunakan teorema Bayes:
$$
P(\text{flu} \ | \ \text{tidak vaksin}) = \frac{n_{\text{flu dan tidak vaksin}}}{n_{\text{tidak vaksin}}}
$$

**Perhitungan:**
$$
\begin{align*}
P(\text{flu} \ | \ \text{tidak vaksin}) &= \frac{n(\text{flu} \cap \text{tidak vaksin})}{n(\text{tidak vaksin})} \\
&= \frac{900}{9900} \\
&\approx 0.0909
\end{align*}
$$
<br>
Jadi, peluang seseorang akan mengidap flu apabila telah divaksin ($P(\text{flu} \ | \ \text{vaksin})$) adalah 0.0476, sedangkan peluang seseorang akan mengidap flu apabila belum divaksin ($P(\text{flu} \ | \ \text{tidak vaksin})$) adalah 0.0909.

<a name='B'></a>
## Case B = Pengaruh Cuaca Terhadap Frekuensi Penggunaan Transportasi Umum

- Telah diamati bahwa cuaca memiliki pengaruh terhadap keputusan orang menggunakan transportasi umum. Anda memiliki data historis dari perusahaan transportasi umum yang mencatat jumlah penumpang per hari selama berbagai kondisi cuaca.

- Diberikan data berikut

| Data Operasi       | Pengguna ketika Hujan | Pengguna ketika Cerah | Promo yang berlaku | Zona Tarif |
|--------------------|-------|-------|--------------------|------------|
| Grab-bike          | 4     | 475   |Diskon 10%          |Zona 1      |
| Grab-car           | 185   | 49    |Tidak ada promo     |Zona 2      |




---
<a name='2'></a>
### **Soal 2**:
A. Berapa peluang **orang menggunakan produk grab-bike?** <br>
B. Berapa peluang **orang menggunakan produk grab-car** jika diketahui **terjadi hujan**<br>
C. Berapa peluang **orang menggunakan produk grab-car** jika diketahui **hari cerah**<br>
D. Bagaimana konsep probabilitas bersyarat digunakan untuk menghitung peluang penggunaan produk grab-car ketika diketahui kondisi cuaca (hujan atau cerah), dan mengapa ini penting dalam menganalisis data operasi seperti ini?

In [ ]:
import numpy as np
import pandas as pd

from scipy.stats import binom

pengguna_hujan_grab_bike = 4
pengguna_cerah_grab_bike = 475
total_grab_bike = pengguna_hujan_grab_bike + pengguna_cerah_grab_bike

pengguna_hujan_grab_car = 185
pengguna_cerah_grab_car = 49
total_grab_car = pengguna_hujan_grab_car + pengguna_cerah_grab_car

# Total pengguna transportasi umum (Grab-bike dan Grab-car)
total_pengguna_transportasi_umum = total_grab_bike + total_grab_car


### A. Berapa peluang **orang menggunakan produk grab-bike?**

In [ ]:
soal_2A = total_grab_bike / total_pengguna_transportasi_umum

print(f"2A. Peluang orang menggunakan produk grab-bike adalah : {soal_2A:.2f}")

2A. Peluang orang menggunakan produk grab-bike adalah : 0.67


### B. Berapa peluang **orang menggunakan produk grab-car** jika diketahui **terjadi hujan**

In [ ]:
Soal_2B = pengguna_hujan_grab_car / (pengguna_hujan_grab_car + pengguna_hujan_grab_bike)

print(f"2B. Peluang orang menggunakan produk go-car jika diketahui terjadi hujan adalah : {Soal_2B:.2f}")

2B. Peluang orang menggunakan produk go-car jika diketahui terjadi hujan adalah : 0.98


Solver:

$$
\begin{align*}
P(\text{A} \ | \ \text{B}) &= \cfrac{P(\text{A} \cap \text{B})}{P(\text{B})}
\end{align*}
$$

### C. Berapa peluang **orang menggunakan produk grab-car** jika diketahui **hari cerah**

In [ ]:
Soal_2C = pengguna_cerah_grab_car / (pengguna_cerah_grab_car + pengguna_cerah_grab_bike)

print(f"2C. peluang orang menggunakan produk grab-car jika diketahui hari cerah : {Soal_2C:.2f}")

2C. peluang orang menggunakan produk grab-car jika diketahui hari cerah : 0.09


### D. Bagaimana konsep probabilitas bersyarat digunakan untuk menghitung peluang penggunaan produk grab-car ketika diketahui kondisi cuaca (hujan atau cerah), dan mengapa ini penting dalam menganalisis data operasi seperti ini?

Konsep probabilitas bersyarat digunakan untuk menghitung peluang penggunaan produk Grab-car ketika kondisi cuacanya (hujan atau cerah) diketahui. Dapat menggunakan rumus probabilitas bersyarat:

$$
P(\text{Grab-car} \ | \ \text{cuaca}) = \frac{P(\text{Grab-car} \cap \text{cuaca})}{P(\text{cuaca})}
$$

Konsep probabilitas bersyarat penting dalam menganalisis data operasi seperti ini karena memungkinkan pemahaman yang lebih baik tentang pengaruh kondisi cuaca terhadap penggunaan produk, seperti Grab-car. Dengan memahami probabilitas penggunaan Grab-car saat cuaca hujan dan cerah, perusahaan transportasi dapat mengoptimalkan strategi operasional, termasuk penyesuaian promosi yang lebih efektif, dan pengembangan strategi lain untuk meningkatkan kepuasan pelanggan serta efisiensi operasional.

**Penjelasan Penyelesaian CASE B** <br>

A. Berapa peluang orang menggunakan produk grab-bike? <br>
B. Berapa peluang orang menggunakan produk grab-car jika diketahui terjadi hujan <br>
C. Berapa peluang orang menggunakan produk grab-car jika diketahui hari cerah <br>
D. Bagaimana konsep probabilitas bersyarat digunakan untuk menghitung peluang penggunaan produk grab-car ketika diketahui kondisi cuaca (hujan atau cerah), dan mengapa ini penting dalam menganalisis data operasi seperti ini?<br>
### **Langkah Penyelesaian**
### **A. Peluang Orang Menggunakan Produk Grab-bike**
1. Menghitung total pengguna Grab-bike pada semua kondisi cuaca:
   Total pengguna Grab-bike = Jumlah pengguna saat hujan + Jumlah pengguna saat cerah
2. Menghitung total pengguna transportasi umum (Grab-bike dan Grab-car).
3. Menghitung peluang orang menggunakan produk Grab-bike:
   Peluang Grab-bike = (Total pengguna Grab-bike) / (Total pengguna transportasi umum)
   <br>
   <br>
$$
\begin{align*}
\text{Total pengguna Grab-bike} &= 479 \\
\text{Total pengguna transportasi umum} &= 713 \\
\text{Peluang Grab-bike} &= \frac{479}{713} \approx 0.6725
\end{align*}
$$
<br>

### **B. Peluang Orang Menggunakan Produk Grab-car Saat Hujan**
1. Menghitung total pengguna Grab-car saat hujan.
2. Menghitung total pengguna Grab-bike dan Grab-car saat hujan.
3. Menghitung peluang orang menggunakan produk Grab-car saat hujan:
   Peluang Grab-car (saat hujan) = (Total pengguna Grab-car saat hujan) / (Total pengguna saat hujan)
   <br>
   <br>
   $$
\begin{align*}
\text{Total pengguna Grab-car saat hujan} &= 185 \\
\text{Total pengguna saat hujan} &= 189 \\
\text{Peluang Grab-car (saat hujan)} &= \frac{185}{189} \approx 0.9788
\end{align*}
$$
<br>

### C. Peluang Penggunaan Produk Grab-car Saat Cuaca Cerah

1. Menghitung total pengguna Grab-car saat cuaca cerah.
2. Menghitung total pengguna Grab-bike dan Grab-car saat cuaca cerah.
3. Menghitung peluang orang menggunakan produk Grab-car saat cuaca cerah:

   Peluang Grab-car (saat cuaca cerah) = (Total pengguna Grab-car saat cuaca cerah) / (Total pengguna saat cuaca cerah)
   
   - Total pengguna Grab-car saat cuaca cerah adalah 49  
   - Total pengguna saat cuaca cerah adalah 524  
   - Peluang Grab-car (saat cuaca cerah) adalah 49 / 524 ≈ 0.0935

### D. Konsep Probabilitas Bersyarat

Menggunakan konsep probabilitas bersyarat untuk menghitung peluang penggunaan produk Grab-car ketika diketahui kondisi cuaca (hujan atau cerah).

- Peluang Grab-car (saat hujan) = (Jumlah pengguna Grab-car saat hujan) / (Total pengguna Grab-car dan Grab-bike saat hujan)
- Peluang Grab-car (saat cuaca cerah) = (Jumlah pengguna Grab-car saat cuaca cerah) / (Total pengguna Grab-car dan Grab-bike saat cuaca cerah)

Di mana:

- \( P(\text{Grab-car} | \text{cuaca}) \) adalah probabilitas bahwa orang akan menggunakan produk Grab-car jika kondisi cuacanya diketahui.
- \( P(\text{Grab-car} \cap \text{cuaca}) \) adalah probabilitas bahwa orang menggunakan produk Grab-car dan kondisi cuacanya adalah hujan atau cerah.
- \( P(\text{cuaca}) \) adalah probabilitas kondisi cuaca, yaitu hujan atau cerah.






---
<a name='C'></a>
## **Case C**: Sebuah perusahaan IT memiliki dua divisi, Divisi Pengembangan dan Divisi Pemasaran, yang keduanya mempekerjakan para profesional IT. Dalam Divisi Pengembangan, 30% dari pekerja memiliki pengalaman kerja kurang dari 3 tahun, sedangkan di Divisi Pemasaran, 70% dari pekerja memiliki pengalaman kerja kurang dari 3 tahun.

<a name='3'></a>
### Soal 3.
A. Jika seorang pekerja dipilih secara acak dan ternyata memiliki pengalaman kerja kurang dari 3 tahun, berapa probabilitas bahwa pekerja tersebut berasal dari Divisi Pengembangan? <br>
B. Dengan mempertimbangkan informasi dari Divisi Pengembangan dan Divisi Pemasaran seperti yang diberikan sebelumnya, jika diketahui bahwa sebuah proyek baru memerlukan tim yang terdiri dari pekerja dengan pengalaman kerja kurang dari 3 tahun, dan satu pekerja dipilih secara acak dari keseluruhan pekerja yang memenuhi kriteria ini, berapa rasio antara probabilitas pekerja tersebut berasal dari Divisi Pengembangan terhadap probabilitas pekerja tersebut berasal dari Divisi Pemasaran?

### A. Jika seorang pekerja dipilih secara acak dan ternyata memiliki pengalaman kerja kurang dari 3 tahun, berapa probabilitas bahwa pekerja tersebut berasal dari Divisi Pengembangan?

In [ ]:
# Mengatur probabilitas bahwa seorang pekerja memiliki pengalaman kerja kurang dari 3 tahun untuk setiap divisi
prob_pengembangan_min3 = 0.3
prob_pemasaran_min3 = 0.7

# Mengatur probabilitas bahwa seorang pekerja berasal dari masing-masing divisi
prob_pengembangan = 0.5
prob_pemasaran = 0.5

# Menghitung probabilitas bahwa seorang pekerja berasal dari setiap divisi dan memiliki pengalaman kurang dari 3 tahun
prob_pengembangan_dan_min3 = prob_pengembangan * prob_pengembangan_min3
prob_pemasaran_dan_min3 = prob_pemasaran * prob_pemasaran_min3
prob_min3 = prob_pengembangan_dan_min3 + prob_pemasaran_dan_min3

# Menghitung probabilitas bahwa seorang pekerja dipilih secara acak dari Divisi Pengembangan jika memiliki pengalaman kurang dari 3 tahun
prob_pengembangan_acak = prob_pengembangan_dan_min3 / prob_min3

Soal_3_A = prob_pengembangan_acak
print(f"3A. Jawaban : {Soal_3_A:.2f}")


3A. Jawaban : 0.30


### B. Rasio probabilitas pekerja berasal dari Divisi Pengembangan terhadap Divisi Pemasaran

In [ ]:
Soal_3_B = prob_pengembangan / prob_pemasaran
print(f"3B. Jawaban : {Soal_3_B:.2f}")

3B. Jawaban : 0.43


**Penjelasan Penyelesaian Soal 3** <br>
Dalam menyelesaikan soal ini, ada beberapa konsep dasar probabilitas yang digunakan:

**Aturan Probabilitas Total:** Digunakan untuk menghitung probabilitas suatu kejadian dengan mempertimbangkan semua kemungkinan kondisi atau kategori yang mungkin. Aturan probabilitas total digunakan untuk menghitung probabilitas bahwa seorang pekerja yang dipilih secara acak berasal dari Divisi Pengembangan, jika diketahui bahwa pekerja tersebut memiliki pengalaman kerja kurang dari 3 tahun.

**Rasio Probabilitas:** Untuk menghitung rasio antara probabilitas bahwa seorang pekerja berasal dari Divisi Pengembangan dibandingkan dengan probabilitas bahwa pekerja tersebut berasal dari Divisi Pemasaran, jika diketahui bahwa pekerja tersebut memiliki pengalaman kerja kurang dari 3 tahun.

### **Langkah Penyelesaian :**

#### ***Soal 3.A: Probabilitas bahwa pekerja tersebut berasal dari Divisi Pengembangan jika memiliki pengalaman kurang dari 3 tahun***

Gunakan aturan probabilitas total:

$$
P(\text{Pengembangan | kurang dari 3 tahun}) = \frac{P(\text{Pengembangan}) \times P(\text{kurang dari 3 tahun | Pengembangan})}{P(\text{kurang dari 3 tahun})}
$$
<br>
Ketika sudah memiliki nilai-nilai probabilitas yang diperlukan:
<br>

$$
\begin{align*}
P(\text{Pengembangan}) &= 0.5 \\
P(\text{kurang dari 3 tahun | Pengembangan}) &= 0.3 \\
P(\text{kurang dari 3 tahun}) &= 0.15 + 0.35 = 0.50 \quad \text{(Total probabilitas pekerja yang memiliki pengalaman kurang dari 3 tahun)}
\end{align*}
$$

Gunakan rumus di atas untuk menghitung probabilitasnya:

$$
\begin{align*}
P(\text{Pengembangan | kurang dari 3 tahun}) &= \frac{0.5 \times 0.3}{0.50} \\
&= \frac{0.15}{0.50} \\
&= 0.30
\end{align*}
$$
Jadi, hasilnya adalah 0.30.
<br>

#### **Soal 3.B: Rasio probabilitas pekerja berasal dari Divisi Pengembangan terhadap Divisi Pemasaran**
Dari soal, diketahui bahwa probabilitas pekerja berasal dari Divisi Pengembangan adalah 0,30 atau 30%. Kemudian, karena total probabilitas harus sama dengan 1 (atau 100%), maka probabilitas pekerja berasal dari Divisi Pemasaran dapat dihitung dengan mengurangkan probabilitas pekerja berasal dari Divisi Pengembangan dari 1.

Jadi, jika probabilitas pekerja berasal dari Divisi Pengembangan adalah 0,30, maka probabilitas pekerja berasal dari Divisi Pemasaran dapat dihitung dengan rumus sebagai berikut.

**Probabilitas pekerja divisi pengembangan :**

$$
P(\text{Divisi M}) = 1 - P(\text{Divisi P}) = 1 - 0,30 = 0,70
$$
<br>

Hitung rasio antara probabilitas pekerja berasal dari Divisi Pengembangan<br>
**Rasio Probabilitas**:
   $$
   \text{Rasio Probabilitas} = \cfrac{P(\text{Pengembangan})}{P(\text{Pemasaran})}
   \cfrac{P(\text{Divisi P})}{P(\text{Divisi M})} = \cfrac{0.30}{0.70} = 0.43
   $$
<br>
Jadi, Ini menunjukkan bahwa kemungkinan lebih tinggi untuk memilih pekerja dengan pengalaman kurang dari 3 tahun dari Divisi Pemasaran daripada dari Divisi Pengembangan.


---
<a name='D'></a>
## **Case D**:
Sebuah perusahaan memproduksi sekrup dan melakukan inspeksi kualitas sekrup. Perusahaan tersebut menyimpan data inspeksi sekrup cacat dalam file format csv yang kita baca di bawah ini

<a name='4'></a>
### **Soal 4**: Berapa peluang sekrup cacat dari data inspeksi perusahaan tersebut <br>

<a name='5'></a>
### **Soal 5**: Berapa proporsi banyak kemasan terjual yang harus diganti oleh perusahaan?

Keterangan
- (1 = Lolos Uji Cacat, 0 = Cacat)
- Setiap kemasan berisi 10 sekrup
- Kemasan harus diganti jika terdapat satu atau lebih sekrup cacat di dalamnya.

In [ ]:
# Pastikan file .csv berada pada directory yang sama
df = pd.read_csv("./soal4_defect_test.csv")
df


,id_sekrup,lolos_uji_cacat
0,1,1
1,2,1
2,3,1
3,4,1
4,5,1
...,...,...
495,496,0
496,497,1
497,498,1
498,499,1


### **Soal 4**: Berapa peluang sekrup cacat dari data inspeksi perusahaan tersebut <br>

In [ ]:
jumlah_sekrup_total = 500 * 10
jumlah_kemasan_cacat = 5
jumlah_sekrup_cacat = jumlah_kemasan_cacat * 10
peluang_cacat = jumlah_sekrup_cacat / jumlah_sekrup_total

# Menampilkan hasil
Soal_4 = peluang_cacat
print("Soal 4: Peluang sekrup cacat dari data inspeksi perusahaan adalah", Soal_4)

Soal 4: Peluang sekrup cacat dari data inspeksi perusahaan adalah 0.01


**Penjelasan Penyelesaian Soal 4** <br>

#### **Langkah langkah :**
1. Membaca Data: Data inspeksi dari file CSV dibaca menggunakan pandas di Python.
2. Menghitung Jumlah Sekrup Total:
   - Jumlah sekrup per kemasan: 10
   - Jumlah kemasan: 500
   - Jumlah sekrup total = 10 * 500 = 5000 sekrup
3. Menghitung Jumlah Sekrup Cacat:
   - Jumlah kemasan cacat: 5
   - Jumlah sekrup per kemasan: 10
   - Jumlah sekrup cacat = 5 * 10 = 50 sekrup
4. Menghitung Peluang Sekrup Cacat: Peluang sekrup cacat dihitung dengan membagi jumlah sekrup cacat dengan jumlah sekrup total.
   - Peluang sekrup cacat = (Jumlah sekrup cacat) / (Jumlah sekrup total) = 50 / 5000 = 0.01



---
<a name='5'></a>
### **Soal 5**: Berapa probabilitas sebuah kemasan perlu diganti, asumsikan bahwa probabilitas sebuah sekrup cacat adalah 0.01 dan setiap kemasan berisi 10 sekrup?

Clue: $P(X>1)$ bisa dicari dengan
$$P(X>1) = 1 - P(X=0) - P(X=1)$$

In [ ]:
jumlah_sekrup_per_kemasan = 10  # Setiap kemasan berisi 10 sekrup
probabilitas_cacat = 0.01  # Probabilitas sebuah sekrup cacat

# Menghitung probabilitas setidaknya satu sekrup cacat dalam satu kemasan
prob_setidaknya_satu_cacat = 1 - binom.pmf(k=0, n=jumlah_sekrup_per_kemasan, p=probabilitas_cacat)

# Menampilkan hasil
Soal_5 = prob_setidaknya_satu_cacat
print(Soal_5)

0.09561792499119592


**Penjelasan Penyelesaian Soal 5** <br>
Langkah-langkah dalam menyelesaikan soal ini:

1. Disini memiliki probabilitas bahwa satu sekrup adalah cacat (probabilitas_cacat).
2. Mencari probabilitas bahwa setidaknya satu dari sepuluh sekrup dalam satu kemasan adalah cacat.
3. Gunakan rumus distribusi binomial kumulatif: 1 - binom.pmf(k=0, n=jumlah_sekrup_per_kemasan, p=probabilitas_cacat). <br>
Di mana:
  - k adalah 0 karena kita ingin setidaknya satu sekrup cacat.
  - n adalah jumlah percobaan (jumlah sekrup per kemasan).
  - p adalah probabilitas keberhasilan dalam satu percobaan (probabilitas sebuah sekrup cacat).

##### **Menggunakan rumus distribusi binomial kumulatif:**
$$P(X \geq 1) = 1 - P(X=0)$$

Di mana $P(X=0)$ adalah probabilitas bahwa tidak ada sekrup cacat dalam satu kemasan.

Untuk menghitung $P(X=0)$, kita menggunakan fungsi probabilitas massa binomial (binom.pmf) dengan
  - $k=0$ (tidak ada sekrup cacat),
  - $n=10$ (jumlah sekrup per kemasan),
  - $p=0.01$ (probabilitas sekrup cacat).
  
Jadi:

$$P(X=0) = \binom{10}{0} \times (0.01)^0 \times (1-0.01)^{10-0}$$
$$P(X=0) = 1 \times 1 \times 0.99^{10}$$
$$P(X=0) = 0.99^{10}$$

<br>
Sekarang, gunakan hasil ini untuk mencari $P(X \geq 1)$:

$$P(X \geq 1) = 1 - 0.99^{10}$$
<br>
Nilai dari $0.99^{10}$, kemudian kurangkan dari 1 untuk mendapatkan probabilitas bahwa setidaknya satu sekrup dalam satu kemasan adalah cacat.

$$P(X \geq 1) = 1 - 0.99^{10} = 1 - 0.9043820750088044 \approx 0.09561792499119592$$
<br>
Jadi, probabilitas bahwa sebuah kemasan perlu diganti adalah sekitar 0.0956 atau sekitar 9.56%.




---
<a name='6'></a>
### **Soal 6**: Sebuah kompetisi internasional mengevaluasi atlet berdasarkan dua kriteria: skor teknik dan skor artistik. Berikut ini adalah 25 total atlet yang berkompetisi. Juri memberikan penilaian untuk masing-masing atlet berdasarkan kedua kriteria tersebut. Data penilaian telah dikumpulkan sebagai berikut:

| Atlet   | Skor Teknik (X) | Skor Artistik (Y) |
|---------|-----------------|-------------------|
| Atlet 1 | 96              | 91                |
| Atlet 2 | 88              | 87                |
| Atlet 3 | 92              | 89                |
| Atlet 4 | 90              | 94                |
| Atlet 5 | 94              | 90                |
| Atlet 6 | 89              | 85                |
| Atlet 7 | 93              | 88                |
| Atlet 8 | 87              | 92                |
| Atlet 9 | 95              | 93                |
| Atlet 10| 91              | 86                |
| Atlet 11| 90              | 91                |
| Atlet 12| 92              | 88                |
| Atlet 13| 89              | 90                |
| Atlet 14| 96              | 95                |
| Atlet 15| 88              | 89                |
|Atlet 20 |	92	            | 91                |
|Atlet 21 |	89	            | 86                |
|Atlet 23 |	87	            | 90                |
|Atlet 24 |	96	            | 95                |
|Atlet 25 |	88	            | 89                |


Hitung <br>
A. Untuk melihat tingkat kesulitan dari perlombaan ini, hitung nilai exspektasi untuk skor teknikal dan artistik nya <br>
B. Untuk mengetahui persebaran nilai dari masing-masing atlet, hitung nilai variance untuk kedua skor tersebut <br>
C. Hitung Covariance Matrix antara skor teknikal dengan skor artistik untuk mengetahui korelasi kedua skor tersebut<br>

**Clue**
- Hati hati dengan kata kata total, apakah itu sample? atau populasi?

In [ ]:
# Solver soal 6
import numpy as np

# Data skor teknikal dan skor artistik
skor_teknik = np.array([96, 88, 92, 90, 94, 89, 93, 87, 95, 91, 90, 92, 89, 96, 88, 92, 89, 87, 96, 88])
skor_artistik = np.array([91, 87, 89, 94, 90, 85, 88, 92, 93, 86, 91, 88, 90, 95, 89, 91, 86, 90, 95, 89])

# A. Expectation (Nilai Harapan)
expectation_teknik = np.mean(skor_teknik)
expectation_artistik = np.mean(skor_artistik)


# B. Variance (Varians)
variance_teknik = sum((x - expectation_teknik)** 2 for x in skor_teknik ) / len(skor_teknik)
variance_artistik = sum((y - expectation_artistik)** 2 for y in skor_artistik ) / len(skor_artistik)


# C. Covariance Matrix (Matriks Kovarians)
covariance_matrix = np.cov(skor_teknik, skor_artistik)

# Menampilkan hasil
print("A. Expectation (Nilai Harapan):")
print("Nilai Harapan Skor Teknik:", expectation_teknik)
print("Nilai Harapan Skor Artistik:", expectation_artistik)
print("\nB. Variance (Varians):")
print("Varians Skor Teknik:", variance_teknik)
print("Varians Skor Artistik:", variance_artistik)
print("\nC. Covariance Matrix (Matriks Kovarians):")
print(covariance_matrix)

A. Expectation (Nilai Harapan):
Nilai Harapan Skor Teknik: 91.1
Nilai Harapan Skor Artistik: 89.95

B. Variance (Varians):
Varians Skor Teknik: 8.989999999999998
Varians Skor Artistik: 7.9475

C. Covariance Matrix (Matriks Kovarians):
[[9.46315789 4.26842105]
 [4.26842105 8.36578947]]


### **Penjelasan Penyelesaian Soal 6**

### **A. Perhitungan nilai harapan (expectation):**

Nilai harapan atau rata-rata dari suatu data dapat dihitung dengan menjumlahkan semua nilai dalam data tersebut, kemudian dibagi dengan jumlah total data.<br>
Untuk skor teknikal:<br>
- Nilai harapan skor teknikal = (96 + 88 + 92 + 90 + 94 + 89 + 93 + 87 + 95 + 91 + 90 + 92 + 89 + 96 + 88 + 92 + 89 + 87 + 96 + 88) / 20
= 1822 / 20
= 91.1<br>

Untuk skor artistik:
- Nilai harapan skor artistik = (91 + 87 + 89 + 94 + 90 + 85 + 88 + 92 + 93 + 86 + 91 + 88 + 90 + 95 + 89 + 91 + 86 + 90 + 95 + 89) / 20
= 1796 / 20
= 89.8

### **B. Perhitungan varians:**

Varians adalah rata-rata dari kuadrat selisih antara setiap nilai dalam data dengan nilai harapan, yaitu rumus ∑(x - μ)² / n.<br>
Untuk skor teknikal:
- Varians skor teknikal = [(96 - 91.1)² + (88 - 91.1)² + ... + (88 - 91.1)²] / 20
= [20.25 + 9.61 + ... + 9.61] / 20
= 179.8 / 20
= 8.989999999999998<br>

Untuk skor artistik:
- Varians skor artistik = [(91 - 89.8)² + (87 - 89.8)² + ... + (89 - 89.8)²] / 20
= [3.24 + 6.76 + ... + 24.01] / 20
= 158.95 / 20
= 7.9475

### **C. Perhitungan matriks kovarians:**

Matriks kovarians mengukur seberapa dua variabel berubah bersamaan. Rumusnya adalah Σ((X - μx) * (Y - μy)) / (n - 1).
Dalam case ini, menggunakan np.cov() untuk menghitung kovarians antara skor teknikal dan skor artistik.
